staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -288,6 +294,40 @@ func GetHistogramVecFromGatherer(gatherer metrics.Gatherer, metricName string, l
 	return vec, nil
 }
 
+type LabeledHistogram struct {
+	Labels    map[string]string
+	Histogram Histogram
+}
+
+// GetHistogramsFromGatherer collects a metric from a gatherer implementing k8s.io/component-base/metrics.Gatherer interface.
+// If the metric is a HistogramVec, each metric from the HistogramVec will be collected with a label map one by one.
+// Used only for testing purposes where we need to gather metrics directly from a running binary (without metrics endpoint).
+func GetHistogramsFromGatherer(gatherer metrics.Gatherer, metricName string, lvMap map[string]string) ([]LabeledHistogram, error) {
+	metricFamily, err := validateMetricFamily(gatherer, metricName)
+	if err != nil {
+		return nil, err
+	}
+
+	var histogramList []LabeledHistogram
+	for _, metric := range metricFamily.GetMetric() {
+		metricLabels := map[string]string{}
+		labelPairs := map[string]string{}
```

Comment:
nit: instantiate a map using `m := make(map[string]string)` is more conventional.